src/analysis/special_functions/gamma.lean

```diff
@@ -519,3 +534,425 @@ end
+      dsimp only,
+      rw fpow hx hq,
+      congr' 1,
+      exact (norm_of_nonneg (posf _ _ x hx)).symm },
+    { refine continuous_on.ae_strongly_measurable _ measurable_set_Ioi,
+      refine (continuous.continuous_on _).mul (continuous_at.continuous_on (λ x hx, _)),
+      { exact continuous_exp.comp (continuous_const.mul continuous_id'), },
+      { exact continuous_at_rpow_const _ _ (or.inl (ne_of_lt hx).symm), } } },
+  have e : (1 / p).is_conjugate_exponent (1 / (1 - p)) := ⟨by rwa [lt_div_iff hp, one_mul],
+    by rw [one_div_one_div, one_div_one_div, ←add_sub_assoc, add_sub_cancel']⟩,
+  convert measure_theory.integral_mul_le_Lp_mul_Lq_of_nonneg  e (posf' p s) (posf' (1 - p) t)
+    (f_mem_Lq hp hs) (f_mem_Lq (by linarith : 0 < 1 - p) ht) using 1,
+  { refine set_integral_congr measurable_set_Ioi (λ x hx, _),
+    dsimp only [f],
+    rw (by { rw ←exp_add, congr' 1, ring} : exp (-x) = exp (-p * x) * exp (-(1-p) * x)),
+    have : x ^ (p * s + (1 - p) * t - 1) = (x ^ (p * (s - 1))) * (x ^ ((1 - p) * (t - 1))),
+    { rw ←rpow_add hx, congr' 1, ring, },
+    rw this,
+    ring },
+  { rw [one_div_one_div, one_div_one_div],
+    congr' 2;
+      refine set_integral_congr measurable_set_Ioi (λ x hx, _);
+      apply fpow hx;
+      linarith },
+end
+
+lemma convex_on_log_Gamma : convex_on ℝ (Ioi 0) (log ∘ Gamma) :=
+begin
+  refine convex_on_iff_forall_pos.mpr ⟨convex_Ioi _, λ x hx y hy p q hp hq hpq, _⟩,
+  have : q = 1 - p := by linarith, subst this,
+  simp_rw [function.comp_app, smul_eq_mul],
+  rw [←log_rpow (Gamma_pos_of_pos hy),
+    ←log_rpow (Gamma_pos_of_pos hx),
+    ←log_mul
+      ((rpow_pos_of_pos (Gamma_pos_of_pos hx) _).ne') (rpow_pos_of_pos (Gamma_pos_of_pos hy) _).ne',
+    log_le_log
+      (Gamma_pos_of_pos (add_pos (mul_pos hp hx) (mul_pos hq hy)))
+      (mul_pos
+        (rpow_pos_of_pos (Gamma_pos_of_pos hx) _) (rpow_pos_of_pos (Gamma_pos_of_pos hy) _))],
+  exact Gamma_mul_add_mul_le_rpow_Gamma_mul_rpow_Gamma hx hy hp (by linarith),
+end
+
+section bohr_mollerup
+
+/-! ## The Euler limit formula and the Bohr-Mollerup theorem
+
+In this section we prove two interelated statements about the `Γ` function on the positive reals:
+
+* the Euler limit formula `real.tendsto_log_gamma_seq`, stating that the sequence
+  `x * log n + log n! - ∑ (m : ℕ) in finset.range (n + 1), log (x + m)`
+  tends to `log Γ(x)` as `n → ∞`.
+* the Bohr-Mollerup theorem (`real.eq_Gamma_of_log_convex`) which states that `Γ` is the unique
+  *log-convex*, positive-real-valued function on the positive reals satisfying
+  `f (x + 1) = x f x` and `f 1 = 1`.
+
+To do this, we prove that any function satisfying the hypotheses of the Bohr--Mollerup theorem must
+agree with the limit in the Gauss formula, so there is at most one such function. Then we show that
+`Γ` satisfies these conditions.
+-/
+
+lemma tendsto_log_comp_add_one_sub_log : tendsto (λ (k : ℕ), log (↑k + 1) - log ↑k) at_top (𝓝 0) :=
```

Comment:
```suggestion
lemma tendsto_log_comp_add_one_sub_log : tendsto (λ (k : ℕ), log (k + 1) - log k) at_top (𝓝 0) :=
```
No need for the explicit coercions: Lean knows that you are applying `log`, so you need a real number.